source/plugins.rst

```diff
@@ -4,6 +4,260 @@
+
+It has a few attributes, all optional.
+
+{|class=wikitable
+!name
+!default
+!effect
+|-
+!type
+|python
+|selects the interpreter (currently only Python is supported)
+|-
+!stdout
+|ignore
+|What to do with the script standard output:
+* **ignore** output is discarded
+* **log** the output is shown in the Synfig log
+* **message** an error message is shown to the user
+|-
+!stderr
+|message
+|Same as above, but for standard error
+|}
+
+Example:
+::
+    <exec>myscript.py</exec>
+
+Changing stream behaviour:
+
+::
+<exec stdout="log" stderr="ignore">myscript.py</exec>
+
+
+<exporter>
+----
+
+Defines a new exporter, used to convert synfig files into other formats.
+
+A plugin can define multiple exporters.
+
+Exporters will be shown in the Export dialog.
+
+The exporter contains the following sub-elements:
+
+* <exec> must have exactly one of these
+* <extension> must have at least one of them
+* <description> must have the default version
+
+Example:
+::
+   <exporter>
+      <extension>svg</extension>
+      <extension>svgz</extension>
+      <description>Scalable Vector Graphics (*.svg, *.svgz)</description>
+      <exec>svg-exporter.py</exec>
+   </exporter>
+
+<importer>
+----
+
+Works the same as <exporter>, but provides script to convert from other formats into synfig.
+
+A plugin can define multiple importers.
+
+Importers will be shown in the Open file dialog.
+
+Example:
+::
+   <importer>
+      <extension>svg</extension>
+      <extension>svgz</extension>
+      <description>Scalable Vector Graphics (*.svg, *.svgz)</description>
+      <exec>svg-exporter.py</exec>
+   </importer>
+
+
+<extension>
+----
+
+For <importer> and <exporter>, which extensions are supported.
+
+Multiple <extension> elements may be present in an importer or exporter (at least one is required)
+
+Example:
+::
+    <extension>svg</extension>
+
+<description>
+----
+
+For <importer> and <exporter>, the text to be shown in the file dialog.
+
+Similarly to <name> this can be translated using xml:lang
+
+Example:
+::
+<description>Scalable Vector Graphics (*.svg, *.svgz)</description>
+
+
+Script Invocation
+====
+
+Plugins Menu
+----
+
+For scripts run when the user clicks on the plugin name in the Plugins menu, synfig will save a copy of the open canvas and pass the path to that file as the argument to the script.
+
+The script can then modify that file and synfig will reload the canvas to reflect any changes.
+
+Exporter
+----
+
+For an exporter, synfig will pass two arguments to the script: the first is the path to a synfig file containing the open canvas; the second is the file name.
+
+Importer
+----
+
+For an importer, synfig will pass the file selected in the open dialog as first argument, and the path to a temporary synfig file as second argument.
+
+Once the script is completed, synfig will load that second file, so the plugin script should populate it appropriately.
+
+
+= Tutorial =
+
+<videoflash>De9UhUkzNiY|425|344</videoflash>
```

Comment:
Please replace it with https://www.youtube.com/watch?v=De9UhUkzNiY

See also instructions on how to embed video - https://github.com/readthedocs/readthedocs.org/issues/879#issuecomment-51875315